### ISSUE TITLE
Neuron Parameters remain unchanged after setting them and also after training them.

### ISSUE THREAD

**Issue by naveedunjum** [lava-nc/lava-dl#320]
When building the network, the neuron parameters that need to be set don't seem to change even after setting different values. 
For example, for the following network: 
```
self.blocks = torch.nn.ModuleList([
                slayer.block.cuba.Dense(neuron_params, 18, 20),
                slayer.block.cuba.Dense(neuron_params, 20, 18)
     ])
```
with
```
neuron_params = {
                'threshold': 0.1,
                'current_decay': 1,
                'voltage_decay': 1,
                'requires_grad': True,     
            }
```
 when checked from inside the network gives:
```
Dense(
  (neuron): Neuron()
  (synapse): Dense(18, 20, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
)
Voltage Parameter containing:
tensor([4096.], requires_grad=True)
Current Parameter containing:
tensor([4096.], requires_grad=True)
Threshold 0.09375
```
We see from the source code the decay is scaled by `1<<12`, so we get 4096. 
But when changing the neuron parameters to 
```
neuron_params = {
                'threshold': 1,
                'current_decay': 10,
                'voltage_decay': 10,
                'requires_grad': True,     
            }
```
we only see the threshold changing inside the network
```
Dense(
  (neuron): Neuron()
  (synapse): Dense(18, 20, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
)
Voltage Parameter containing:
tensor([4096.], requires_grad=True)
Current Parameter containing:
tensor([4096.], requires_grad=True)
Threshold 1.0
```
The voltage and current decay remain the same.

After training the network with SpikeTime Loss(Oxford tutorial), we again see only the threshold changing.
```Dense(
  (neuron): Neuron()
  (synapse): Dense(18, 20, kernel_size=(1, 1, 1), stride=(1, 1, 1), bias=False)
)
Voltage Parameter containing:
tensor([4096.0020], requires_grad=True)
Current Parameter containing:
tensor([4096.0020], requires_grad=True)
Threshold 0.09375
```

1. When I run this code (add code or minimum test case) ...
```python
def my_code():
    pass
```
2. I get this error ...
```
Error...
```

**Expected behavior**

**Screenshots**
If applicable, add screenshots to help explain your problem. Remove section otherwise.

**Environment (please complete the following information):**
 - Device: [e.g. Laptop, Intel cloud]
 - OS: [e.g. Linux]
 - Lava version [e.g. 0.6.1]

**Additional context**
Add any other context about the problem here. Remove section otherwise.

**Comment by bamsumit** [comment]
1. `current_decay` and `voltage_decay` need to be between 0 and 1. So a value of 10 is most likely getting clamped in this valid range.
2. Thresholds in CUBA-LIF neurons are not learnable.
